src/SFA.DAS.Assessor.Functions/Domain/Print/Services/BatchService.cs

```diff
@@ -41,53 +49,147 @@ public async Task<Batch> Get(int batchNumber)
             return null;
         }
 
-        public async Task<int> NextBatchId()
+        public async Task<int?> BuildPrintBatchReadyToPrint(DateTime scheduledDate, int maxCertificatesToBeAdded)
         {
-            var response = await _assessorServiceApiClient.GetCurrentBatchLog();
+            var nextBatchNumberReadyToPrint = await _assessorServiceApiClient.GetBatchNumberReadyToPrint();
+            if (await _assessorServiceApiClient.GetCertificatesReadyToPrintCount() > 0)
```

Comment:
should we read this number of certificates once into a variable otherwise it could be a moving target in the below do while??

The 2 async calls could be done in parallel then

Reply:
The method is checking for a previous batch which has not been printed, if there are any additional certificates which are not yet in a batch then they will be added to this batch; or if there are certificates to be printed and there is no current batch - create a batch one - then it adds all the certificates which are not in a batch either to the existing or the new batch - so it is a moving target - I will refactor to make this a lot clearer